File: packages/typescript/src/endpoints/ai_bulk_operations.ts
/**
 * Ai_Bulk_Operations API Client
 */

import { AxiosInstance } from 'axios'

export class Ai_Bulk_OperationsClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * DELETE /bulk/delete
   */
  async deleteBulkdelete(analysis_ids?: string[]): Promise<Dict[str, Any]> {
        const params = {}
        const data = analysis_ids
        return this.httpClient.delete(`/bulk/delete`, data, params)
  }
  /**
   * PUT /bulk/update-status
   */
  async putBulkupdatestatus(analysis_updates?: Dict[str, Any][]): Promise<Dict[str, Any]> {
        const params = {}
        const data = analysis_updates
        return this.httpClient.put(`/bulk/update-status`, data, params)
  }
  /**
   * GET /bulk/export
   */
  async getBulkexport(analysis_ids?: string[], export_format?: string): Promise<Any> {
        const params = {'analysis_ids: analysis_ids, export_format: export_format'}
        const data = undefined
        return this.httpClient.get(`/bulk/export`, params)
  }
}

File: packages/python/src/endpoints/ai_bulk_operations.py
"""
Ai_Bulk_Operations API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Ai_Bulk_OperationsClient:
    """Client for ai_bulk_operations endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def delete_bulk_delete(self, analysis_ids: Optional[List[str]] = None):
        """DELETE /bulk/delete"""
        url = f"/bulk/delete"
        params = {}
        data = analysis_ids
        return await self._request("DELETE", url, params=params, json=data)


    async def put_bulk_update_status(self, analysis_updates: Optional[List[Any]] = None):
        """PUT /bulk/update-status"""
        url = f"/bulk/update-status"
        params = {}
        data = analysis_updates
        return await self._request("PUT", url, params=params, json=data)


    async def get_bulk_export(self, analysis_ids: Optional[List[str]] = None, export_format: Optional[str] = None):
        """GET /bulk/export"""
        url = f"/bulk/export"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

